sdk/cosmosdb/cosmos/src/globalEndpointManager.ts

```diff
@@ -183,6 +187,45 @@ export class GlobalEndpointManager {
     }
   }
 
+  private refreshStaleUnavailableLocations(): void {
+    const now = Date.now();
+    this.updateLocation(now, this.unavailableReadableLocations, this.readableLocations);
+    this.unavailableReadableLocations = this.cleanUnavailableLocationList(
+      now,
+      this.unavailableReadableLocations
+    );
+
+    this.updateLocation(now, this.unavailableWriteableLocations, this.writeableLocations);
+    this.unavailableWriteableLocations = this.cleanUnavailableLocationList(
+      now,
+      this.unavailableWriteableLocations
+    );
+  }
+  private updateLocation(now: number, unavailableLocations: Location[], allLocations: Location[]) {
```

Comment:
Please add comments on what this function does.

Reply:
addressed